### DISCUSSION TITLE
Whether the design of delayed messages should be optimized

### DISCUSSION THREAD

**Discussion by crossoverJie** [apache/pulsar#18513] [Q&A] [answered]
When a delayed message is sent that has not yet been acknowledged, a `individuallyDeletedMessages` is created at this point.

What if it takes a long time, say a year, at which point the message on this topic will be slow to be released?

Should we store normal messages and delayed messages separately?

**Comment by lhotari** [comment, marked as answer]
You might want to check https://github.com/apache/pulsar/issues/16763 and https://github.com/apache/pulsar/issues/15500

**Reply by crossoverJie** [reply]
Got it, thx.
